library/src/main/java/com/novoda/downloadmanager/MigrationJob.java

```diff
@@ -218,4 +219,38 @@ private void deleteVersionOneDatabase(InternalMigrationStatus migrationStatus, S
         database.deleteDatabase();
     }
 
+    private void migrateV1FilesToV2Location(FilePersistence filePersistence, Migration migration) {
+        for (Migration.FileMetadata fileMetadata : migration.getFileMetadata()) {
+            filePersistence.create(fileMetadata.newFileLocation(), fileMetadata.fileSize());
+            FileInputStream inputStream = null;
+            try {
+                // open the v1 file
+                inputStream = new FileInputStream(new File(fileMetadata.originalFileLocation().path()));
+                byte[] bytes = new byte[RANDOMLY_CHOSEN_BUFFER_SIZE_THAT_SEEMS_TO_WORK];
+
+                // read the v1 file
+                int readLast = 0;
+                while (readLast != -1) {
+                    readLast = inputStream.read(bytes);
+                    if (readLast != 0 && readLast != -1) {
+                        // write the v1 file to the v2 location
+                        filePersistence.write(bytes, 0, readLast);
+                        bytes = new byte[RANDOMLY_CHOSEN_BUFFER_SIZE_THAT_SEEMS_TO_WORK];
```

Comment:
I might be wrong, but you shouldn't need to recreate this every time.
Its content is overwritten anyway in `inputStream.read(bytes)`, and you are writing only the freshest content (the new `readLast` bytes)

Reply:
I think this is done to avoid the scenario where:

write one -> 200 bytes
write two -> 100 bytes

but I have 100 bytes that were written previously, so it will do the `100 bytes` PLUS whatever is the difference of the first, the missing `100 bytes`. Does that make sense?

Reply:
I don't think that will happen, because let's say 200 bytes is your buffer size.
Then you have 

1. readLast = 200 -> write bytes[0..199]
2. readlast = 100 -> bytes[100..199] contains outdated dat but we only write bytes[0..99]

Reply:
😄 on second look I agree.